internal/core/items/text.rs

```diff
@@ -356,18 +355,21 @@ impl Item for TextInput {
 
                 return InputEventResult::GrabMouse;
             }
-            MouseEvent::Pressed { position, button: PointerEventButton::Middle, .. } => {
-                let clicked_offset = self.byte_offset_for_position(position, window_adapter) as i32;
-                self.as_ref().anchor_position_byte_offset.set(clicked_offset);
-                if !self.has_focus() {
-                    WindowInner::from_pub(window_adapter.window()).set_focus_item(self_rc);
-                }
-                self.set_cursor_position(clicked_offset, true, window_adapter, self_rc);
-                self.paste_clipboard(window_adapter, self_rc, Clipboard::SelectionClipboard);
+            MouseEvent::Pressed { .. } => {
+                #[cfg(not(target_os = "android"))]
+                self.ensure_focus_and_ime(window_adapter, self_rc);
             }
             MouseEvent::Released { button: PointerEventButton::Left, .. } => {
                 self.as_ref().pressed.set(0);
                 self.copy_clipboard(Clipboard::SelectionClipboard);
+                #[cfg(target_os = "android")]
+                self.ensure_focus_and_ime(window_adapter, self_rc);
+            }
+            MouseEvent::Released { position, button: PointerEventButton::Middle, .. } => {
+                let clicked_offset = self.byte_offset_for_position(position, window_adapter) as i32;
+                self.as_ref().anchor_position_byte_offset.set(clicked_offset);
+                self.set_cursor_position(clicked_offset, true, window_adapter, self_rc);
```

Comment:
I think the previous code set the focus. Maybe this should still be done on middle click?

```suggestion
                self.ensure_focus_and_ime(window_adapter, self_rc);
                self.set_cursor_position(clicked_offset, true, window_adapter, self_rc);
```

Reply:
the focus is given on press for the middle click

Reply:
Oops indeed